### PR TITLE
Fold a linked list in Java

### LINKED ISSUE #6438: Fold A Linked List in Java
## 💥 Proposal

The function is expected to place last element after 1st element, 2nd last element after 2nd element and so on.

Example 1
1->2->3->4->5
will fold as
1->5->2->4->3

Example 2
1->2->3->4->5->6
1->6->2->5->3->4

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

### PR DESCRIPTION
**Pull request by Prerna-07**
### Have you read the [Contributing Guidelines on Pull Requests]Yes

Yes

### Description

The function is expected to place last element after 1st element, 2nd last element after 2nd element and so on

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

closes :  #6438

### CODE REVIEW

**Comment by Prerna-07** [comment]
@ankitaggarwal23 , @bhav09 ,@anushkrishnav, @gat786 @atarax665   pls review the PR

File: Java/ds/Fold A Linked List.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+ /*Example 1
+1->2->3->4->5
+will fold as
+1->5->2->4->3 */
+
+/* Example 2
+1->2->3->4->5->6
+1->6->2->5->3->4 */
+// Implement on singly Linked List , using Recurrsion.
+
```
Comment (ankitaggarwal23) [inline Java/ds/Fold A Linked List.java]: Remove redundant description

File: Java/ds/Fold A Linked List.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+*/
+
+/*
+Sample Input/Output:
+Input:  6
+        2 4 5 6 7 8 
+Output: 2 8 4 7 5 6 
+Time Complexity: O(n) 
+Space Complexity: O(n) 
+*/
```
Comment (ankitaggarwal23) [inline Java/ds/Fold A Linked List.java]: one is enough

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Few changes: 

- Add Multi-Line Comments 
- Add a beginner-friendly description at the top
- Use proper code Format 
- Remove extra lines of code and spaces
- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code   
- Add a new line at the EOF
- Use proper display message how you are reading and writing outputs
- Add input-output prompts to make the program more interactive.
- comments you add are above the line of code and not alongside it.

**Comment by Prerna-07** [comment]
@ankitaggarwal23 done sir pls check.

**Pull request merged**